shared/src/commonMain/kotlin/com/example/tasky/agenda/domain/AgendaRepository.kt

```diff
@@ -6,12 +6,23 @@ import com.example.tasky.agenda.data.mapper.toRemoteReminder
 import com.example.tasky.agenda.data.mapper.toRemoteTask
 import com.example.tasky.agenda.data.mapper.toUpdateEventBody
 import com.example.tasky.agenda.domain.model.AgendaItem
+import com.example.tasky.agenda.domain.model.Attendee
 import com.example.tasky.agenda.domain.model.Event
+import com.example.tasky.agenda.domain.model.RemindAtType
 import com.example.tasky.agenda.domain.model.Reminder
 import com.example.tasky.agenda.domain.model.Task
 import com.example.tasky.common.data.model.BaseError
 import com.example.tasky.common.domain.model.ResultWrapper
 import com.example.tasky.common.domain.model.map
+import com.example.tasky.database.database
+import com.example.tasky.database.mapper.toEventEntity
+import com.example.tasky.database.mapper.toReminderEntity
+import com.example.tasky.database.mapper.toTaskEntity
+import com.example.tasky.login.domain.manager.SessionManager
+import kotlinx.datetime.Clock
+import kotlin.time.DurationUnit
+import kotlin.uuid.ExperimentalUuidApi
+import kotlin.uuid.Uuid
 
 interface IAgendaRepository {
```

Comment:
You’ve mixed data and domain layers here. Your interface belongs in domain while the implementation belongs in a separate file in the data layer 🙂